src/Nox.Generator/Infrastructure/Persistence/DbContext.template.cs

```diff
@@ -116,17 +117,20 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
         modelBuilder.ForEntitiesOfType<IEntityConcurrent>(
             builder => builder.Property(nameof(IEntityConcurrent.Etag)).IsConcurrencyToken());
-
-        {{- for entityAtt in enumerationAttributes #Setup Entity Enumerations}}
-        {{- for enumAtt in entityAtt.Attributes}}
-            ConfigureEnumeration(modelBuilder.Entity("{{codeGeneratorState.DomainNameSpace}}.{{enumAtt.EntityNameForEnumeration}}"));
-            {{- if enumAtt.Attribute.EnumerationTypeOptions.IsLocalized}}
-            var enumLocalizedType = _clientAssemblyProvider.GetType("{{codeGeneratorState.DomainNameSpace}}.{{enumAtt.EntityNameForLocalizedEnumeration}}")!;
-            var enumType = _clientAssemblyProvider.GetType("{{codeGeneratorState.DomainNameSpace}}.{{enumAtt.EntityNameForEnumeration}}")!;
-            ConfigureEnumerationLocalized(modelBuilder.Entity(enumLocalizedType), enumType, enumLocalizedType);
-            {{- end }}
-        {{- end }}
-        {{- end }}
+    }
+    
+    private void ConfigureEnumeratedAttributes(ModelBuilder modelBuilder, Entity entity)
+    {
+        foreach(var enumAttribute in entity.Attributes.Where(attribute => attribute.Type == NoxType.Enumeration))
+            {
+                ConfigureEnumeration(modelBuilder.Entity($"{{codeGeneratorState.DomainNameSpace}}.{_codeGenConventions.GetEntityNameForEnumeration(entity.Name, enumAttribute.Name)}"), enumAttribute.EnumerationTypeOptions!);
+                if (enumAttribute.EnumerationTypeOptions!.IsLocalized)
+                {
+                    var enumLocalizedType = _clientAssemblyProvider.GetType($"{{codeGeneratorState.DomainNameSpace}}.{_codeGenConventions.GetEntityNameForEnumerationLocalized(entity.Name, enumAttribute.Name)}")!;
+                    var enumType = _clientAssemblyProvider.GetType($"{{codeGeneratorState.DomainNameSpace}}.{_codeGenConventions.GetEntityNameForEnumeration(entity.Name, enumAttribute.Name)}")!;
+                    ConfigureEnumerationLocalized(modelBuilder.Entity(enumLocalizedType), enumType, enumLocalizedType, enumAttribute.EnumerationTypeOptions!, _noxSolution.Application?.Localization?.DefaultCulture ?? "en-US"); // TODO check if it is not defined if we want to use en-US or just skip seeding localized data
```

Comment:
You don't need to have null-checks for the DefaultCulture, because it is already handled inside the Localization class.

Reply:
Fixed, thanks